omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb

```diff
@@ -626,7 +626,26 @@ def generate_config(node_name)
         gen_ldap
       end
 
-      generate_hash
+      # Removing nils here is required. If any nil values
+      # end up in the hash, Chef will delete attribute.
+      # This seems to happen when we specify attributes
+      # in the dna.json. 'private_chef' and all it's children
+      # start being interpreted as normal attributes, and in
+      # this context, Chef is interpreting nil as delete. When
+      # the attribute was a default attribute, nil was being
+      # interpreted as take the default.
+      remove_nils(generate_hash)
```

Comment:
I might need to get some of your time this week to understand what's driving this one better. It may be that we're better off by setting the attirbute in a way other than dna.json if that's what triggers the behavior.